core/trino-parser/src/main/java/io/trino/sql/tree/SaveMode.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.tree;
+
+public enum SaveMode {
+    IGNORE,
+    REPLACE,
+    FAIL;
+
+    public static SaveMode create(boolean replace, boolean notExists)
```

Comment:
When building AST, we usually just determine the enum values in the `AstBuilder` instead of having a method to resolve them. This is just a minor codestyle comment. IMO the code is easier to read if you see the values inlined.

Reply:
I indeed forgot about CTAS, so this code is used in two places. I have put the code as a static method `toSaveMode` in `AstBuilder`. Let me know if you like that.